src/apps/cardano/address.py

```diff
@@ -16,12 +25,24 @@ def derive_address_and_node(keychain, path: list):
     address_data = [address_root, address_attributes, address_type]
     address_data_encoded = cbor.encode(address_data)
 
-    address = base58.encode(
-        cbor.encode(
-            [cbor.Tagged(24, address_data_encoded), crc.crc32(address_data_encoded)]
-        )
-    )
-    return (address, node)
+    return (_encode_address_raw(address_data_encoded), node)
+
+
+def validate_address(address) -> bool:
+    try:
+        address_hex = base58.decode(address)
+        address_unpacked = cbor.decode(address_hex)
+    except ValueError as e:
+        if __debug__:
+            log.exception(__name__, e)
+        return False
+
+    if not isinstance(address_unpacked, list) or len(address_unpacked) != 2:
+        return False
+
+    address_data_encoded = address_unpacked[0]
```

Comment:
We should also check that `address_data_encoded` is a byte array

Reply:
fixed